src/main/java/com/klarna/hiverunner/builder/HiveShellBuilder.java

```diff
@@ -61,18 +61,24 @@ public void addResource(String targetFile, String data) throws IOException {
         resources.add(new HiveResource(targetFile, data));
     }
 
-    public void setScriptsUnderTest(List<Path> scripts, Charset charset) {
-        int index = 0;
-        for (Path path : scripts) {
-            Preconditions.checkState(Files.exists(path), "File %s does not exist", path);
-            try {
-              String sqlText = new String(Files.readAllBytes(path), charset);
-              scriptsUnderTest.add(new HiveRunnerScript(index++, path, sqlText));
-            } catch (IOException e) {
-                throw new IllegalArgumentException("Failed to load script file '" + path + "'");
-            }
+    public void setScriptsUnderTest(List<Path> scriptPaths, Charset charset) {
+      scriptsUnderTest = fromScriptPaths(scriptPaths, charset);
+    }
+    
+    public List<Script> fromScriptPaths(List<Path> scriptPaths, Charset charset){
+    List<Script> scripts = scriptsUnderTest;
```

Comment:
```suggestion
    List<Script> scripts = new ArrayList();
```
Let's rather do this to avoid this method having a side effect of altering the `scriptsUnderTest` member variable.